src/main/java/io/crate/testing/CrateTestCluster.java

```diff
@@ -273,15 +287,65 @@ protected void before() throws Throwable {
         }
     }
 
+    private void downloadCrate() throws IOException {
+        Files.createDirectories(TMP_CACHE_DIR);
+        if (TMP_WORKING_DIR.equals(workingDir)) {
+            Files.createDirectories(TMP_WORKING_DIR);
+        } else {
+            Files.createDirectories(workingDir);
+        }
+
+        Path crateTmpPath = TMP_CACHE_DIR.resolve(fileNameFromDownloadSource(downloadSource));
+        Path cratePath = crateWorkingDirPath();
+
+        if (Files.exists(crateTmpPath)) {
```

Comment:
use a dedicated .part file while downloading in order to ensure that another process is not writing to it